### ISSUE TITLE
Resource list in sync panel is jumpy on click

### ISSUE THREAD

**Issue by crenshaw-dev** [argoproj/argo-cd#10887]
**Describe the bug**

When I uncheck the first item in the sync panel resource list, a warning appears, pushing all the list items down. If a user has scrolled far down the list, they won't see the warning, but they will feel the jump.

The warning is small. I'd propose that we leave the necessary space for the warning when all resources are checked, and then make the warning visible after the first item is unchecked.

**To Reproduce**

1. Create or open an application in the Argo CD UI
2. In the main Applications page, click the Sync button on the application's card
3. In the resources list at the bottom of the sync panel, uncheck any resource
4. Notice the jump

**Expected behavior**

I'd expect the list to remain still while I check and uncheck items.

**Screenshots**

https://user-images.githubusercontent.com/350466/194935637-2b675df6-03b6-4008-8294-782e9f70958c.mov

**Version**

2.5.0-rc1

**Comment by JessieTeng89** [comment]
@crenshaw-dev I would like to take this issue, could you please assign it to me?
By my understanding, if one of list resources is unchecked, the warning should be displayed.
Do you want to fix it like below screenshots? We will have a empty row even all items are checked.
![image](https://user-images.githubusercontent.com/101035990/195038989-e9d4ac4a-96b8-4b75-b022-5dd39574a973.png)
![image](https://user-images.githubusercontent.com/101035990/195039058-29e1ea1e-53e9-4aeb-9447-0c691b8e6de4.png)

**Comment by crenshaw-dev** [comment]
@JessieTeng89 yep, that's my preference! @rbreeze what do you think? Is the awkward empty row worth it to solve the jumpiness?

**Comment by rbreeze** [comment]
I agree, I think it's worth having an empty row to solve this. Another thought: what if we move the warning to the bottom of the list? That way, there's no empty row and it also doesn't jump.

**Comment by crenshaw-dev** [comment]
For short lists, I think the bottom is fine. For very long lists, maybe the top makes a bit more sense? But very long lists are just kinda bad either way, the user is not going to see the warning if they've scrolled to the middle of the list.

**Comment by rbreeze** [comment]
good point- We can leave a gap at the top then!

**Comment by JessieTeng89** [comment]
@crenshaw-dev @rbreeze 
Thanks for your confirmations, I will start to work on it and create a PR later for your review

**Comment by JessieTeng89** [comment]
Hi @crenshaw-dev 
I created a [PR](https://github.com/argoproj/argo-cd/pull/10970) for this issue, kindly help to review it.
